packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -455,3 +466,71 @@ function defaultGenerateClientInfo({ request }: GraphQLRequestContext) {
     return {};
   }
 }
+
+// Creates trace details from request variables, given a policy for modifying
+// values of private or sensitive variables.
+// The details include the variables in the request, and the privacy enforcer type.
+// If enforcePrivateVariables is a bool or Array, it will act similarly to
+// to the to-be-deprecated options.privateVariables, except that the redacted variable
+// names will still be visible in the UI when 'true.'
+// If enforcePrivateVariables is null, the policy will default to the 'true' case.
+export function makeTraceDetails(
+  variables: Record<string, any>,
+  enforcePrivateVariables?:
```

Comment:
Not doing a full review right now, but my top-level feedback is that I think this would be more valuable if there was access in the context of the enforcer function to the actual operation (AST) that was sent in, and maybe even to the schema as well. In its current form, this doesn't really enable users to, for example, match the variable names that are passed in at this level to the input arguments they represent in the operation.